third_party/winuwp_h264/H264Decoder/H264Decoder.cc

```diff
@@ -334,11 +342,13 @@ HRESULT WinUWPH264DecoderImpl::FlushFrames(uint32_t rtp_timestamp,
       // Convert NV12 to I420. Y and UV sections have same stride in NV12
       // (width). The size of the Y section is the size of the frame, since Y
       // luminance values are 8-bits each.
-      libyuv::NV12ToI420(src_data, width, src_data + (width * height), width,
-                         buffer->MutableDataY(), buffer->StrideY(),
-                         buffer->MutableDataU(), buffer->StrideU(),
-                         buffer->MutableDataV(), buffer->StrideV(), width,
-                         height);
+      const int src_stride_y = buffer_width;
+      const int src_stride_uv = buffer_width;
+      const uint8_t* src_uv = src_data + (src_stride_y * buffer_height);
+      libyuv::NV12ToI420(
```

Comment:
>libyuv::NV12ToI420 [](start = 6, length = 18)

Do you know if Media Foundation have similar capability? I don't remember if a MFT is required for this kind of conversion. It is ok if not, I'm just thinking on decreasing the amount of dependencies.

Reply:
I am pretty sure the MFT decoder can output I420 directly. However I don't want to make that change here because I am slightly concerned as to why this was not done in the first place, and so I am wondering if I am missing some context here. So trying to keep the amount of change to a minimum for now.

Note that `libyuv` is a dependency of `webrtc.lib`, used in various places already, so there's no additional dependency specifically added by this code.

Reply:
The H.264 decoder can only output I420 directly when not using DXVA.   When DXVA is enabled, format support is limited to NV12.